contracts/staking/contracts/src/staking_pools/MixinCumulativeRewards.sol

```diff
@@ -57,27 +57,6 @@ contract MixinCumulativeRewards is
         return cumulativeReward.denominator != 0;
     }
 
-    /// @dev Tries to set a cumulative reward for `poolId` at `epoch`.
-    /// @param poolId Unique Id of pool.
-    /// @param epoch Epoch of cumulative reward.
-    /// @param value Value of cumulative reward.
-    function _trySetCumulativeReward(
-        bytes32 poolId,
-        uint256 epoch,
-        IStructs.Fraction memory value
-    )
-        internal
-    {
-        // Do nothing if it's in the past since we don't want to
-        // rewrite history.
-        if (epoch < currentEpoch
```

Comment:
This was originally:
```
    if (_isCumulativeRewardSet(_cumulativeRewardsByPool[poolId][epoch])) {
        return
    }
```

The intention was: if there is a value then do not overwrite it.

The current version:
```
if (epoch < currentEpoch
            && _isCumulativeRewardSet(_cumulativeRewardsByPool[poolId][epoch]))
    {
        return;
    }
```

The intention here seems to be: it's okay to set historical values, so long as we aren't overwriting anything. And it's always okay to update the value for the current or a future epoch. Is this the intended behavior @dorothy-zbornak ?

Reply:
Doesn't the latter imply that we *can't* set historical values?
This function is gone, though, so can you point to where you derived the new conditions?

From what I can tell, it doesn't look like we ever set a CR at any epoch other than `currentEpoch`.

Reply:
This PR removes this part of the condition: `epoch < currentEpoch`. That was _always_ evaluating to `false`, so this function would never return early.

Reply:
Nvm I see the issue is that this `return` never gets triggered by an existing codepath. So, in theory, this could corrupt state if we finalized a pool and then carried out some action in the same epoch that overwrote the value. 

That said, the function that calls this overwrites using the most recent value. So, this effectively performs a NOP if the CR is already set.

So, at present, this bug is fairly benign and manifests by wasting gas.